pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go

```diff
@@ -966,35 +977,76 @@ func (t *Test) choosePreviousReleases() ([]*clusterupgrade.Version, error) {
 		return []*clusterupgrade.Version{pred, predPred}, nil
 	}
 
+	// possibleUpgradePathsMap maps a version to the possible upgrade paths that
+	// can be taken with exactly n upgrades.
+	possibleUpgradePathsMap := make(map[*clusterupgrade.Version]map[int][][]*clusterupgrade.Version)
+	// findPossibleUpgradePaths finds all legal upgrade paths ending at version `v` with
+	// exactly `numUpgrades` upgrades.
+	var findPossibleUpgradePaths func(v *clusterupgrade.Version, numUpgrades int) ([][]*clusterupgrade.Version, error)
```

Comment:
Nit: `var` declaration seems redundant

Reply:
The `var` declaration is needed for the recursion to work since it's a function literal.

Reply:
Ah, I see. Same as `letrec` vs `let`.